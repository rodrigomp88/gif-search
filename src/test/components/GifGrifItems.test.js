const { shallow } = require("enzyme/build")
const { GifGridItem } = require("../../components/GifGridItem")


describe('Pruebas en GifGrifdItems', () => {

    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg'
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('Debe de mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('debe de tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title)
    })
    test('debe tener la imagen igual al url y al de los props', () => {
        const img = wrapper.find('img');
        // console.log(src);
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })

    test('debe tener animate__animated ', () => {
        const div = wrapper.find('div');
        const className = div.prop('className')

        expect(className.includes('animate__animated')).toBe(true);
    })
    
    
    
    
})
