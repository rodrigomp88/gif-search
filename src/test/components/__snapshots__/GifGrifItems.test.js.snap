// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrifdItems Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce animate__delay-2s"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
